Code-Sleep-Python/Prime/sieve_of_eratosthenes.py

```diff
@@ -0,0 +1,157 @@
+    sq_num = (math.floor(math.sqrt(num)) + 1)**2
+dim = int(sq_num**0.5)  # number of rows and columns of the window
+
+# creating a window having 'dim' rows and columns. Each cell has width of 32px
+width = 32 * dim
+display = pygame.display.set_mode((width, width))
+display.fill(WHITE)  # fill the display with WHITE color
+
+for i in range(0, width, 32):
+    pygame.draw.line(display, BLACK, (i, 0), (i, width)
+                     )  # drawing row lines 32px apart
+for j in range(0, width, 32):
+    # drawing column lines 32px apart
+    pygame.draw.line(display, BLACK, (0, j), (width, j))
+pygame.display.update()
+
+
+def text_objects(text, font):
+    '''
+    Function to create TextSurface and TextRect objects for
+    displaying text on the PyGame window.
+    It takes 2 parameters:
+        text : the text to be displayed
+        font : pygame.font.Font object in which the text is to be rendered
+    '''
+    textSurface = font.render(text, True, BLACK)
+    return textSurface, textSurface.get_rect()
+
+
+def message_display(text, x, y, size):
+    '''
+    Function to display the given 'text' message
+    at a position of (x,y) in size 'size'
+    '''
+    largeText = pygame.font.Font('freesansbold.ttf', size)
+    TextSurf, TextRect = text_objects(text, largeText)
+    TextRect.center = (x, y)
+    display.blit(TextSurf, TextRect)
+    pygame.display.update()
+
+
+celldict = {}  # dictionary to map numbers to their respective cells
+
+
+class Cell:  # creating a class for Cell objects on the grid
+    def __init__(self, x, y, n):
+        self.pos = (x, y)
+        self.value = n
+        self.highlighted = False
+        # adding the number and cell pair to the celldict dictionary
+        celldict[self.value] = self
+        # display the value of the cell on the grid at the center of the cell
+        message_display(f"{self.value}", x + 16, y + 16, 17)
+
+    def highlight(self):
+        if not self.highlighted:
+            x, y = self.pos
+            # create a MAUVE rectangle on top of the cell to highlight
+            # composite numbers
+            pygame.draw.rect(display, MAUVE, (x, y, 32, 32))
+            # rewrite the value of the cell on top of the MAUVE highlight
+            message_display(f"{self.value}", x + 16, y + 16, 17)
+            # to avoid time to re-highlight cells that have already been
+            # highlighted
+            self.highlighted = True
+            # for the effect of moving highlight or "scanning" the grid
+            pygame.time.wait(50)
+
+
+count = 1
+for i in range(0, width, 32):
+    # to deal with internal PyGame events in the event queue which cause the
+    # system to freeze
+    pygame.event.pump()
+    for j in range(0, width, 32):
+        if (count <= num):
+            # create a Cell object at every grid intersection
+            cell = Cell(j, i, count)
+            clock.tick()
+            count += 1
+        else:
+            break
+pygame.display.update()

```

Comment:
why update method here?

Reply:
The update() method was called here in order to show the Cells on the pygame window. Without it, the Cells would not have appeared until the next call of update()